src/components/DialogBox/DialogBox.vue

```diff
@@ -0,0 +1,191 @@
+<script setup lang="ts">
+	import useCustomizableOptions, { type CustomizableOptions } from '@/composables/useCustomizableOptions'
+	import { mdiClose } from '@mdi/js'
+	import { ref } from 'vue'
+	import type { VDialog } from 'vuetify/components'
+	import { config } from './config'
+	import { locales } from './locales'
+
+	const props = withDefaults(defineProps<{
+		title?: string
+		width?: string
+		cancelBtnText?: string
+		confirmBtnText?: string
+		hideActions?: boolean
+		persistent?: boolean
+	} & CustomizableOptions>(), {
+		title: undefined,
+		width: '800px',
+		cancelBtnText: locales.cancelBtn,
+		confirmBtnText: locales.confirmBtn,
+		hideActions: false,
+		persistent: false,
+	})
+
+	defineEmits(['cancel', 'confirm', 'update:modelValue'])
+	defineSlots<{
+		default?: () => undefined
+		title?: () => undefined
+		actions?: () => undefined
+	}>()
+
+	const dialog = defineModel<boolean>({
+		default: false,
+	})
+
+	const dialogContent = ref<VDialog | undefined>(undefined)
+
+	const options = useCustomizableOptions(config, props)
+
+	const closeIcon = mdiClose
+
+	async function getSelectableElements(): Promise<HTMLElement[]> {
+		const parentNode = dialogContent?.value?.$el // Is undefined when dialog is closed
+
+		if (!parentNode) {
+			return []
+		}
+
+		const elements = Array.from(
+			parentNode.querySelectorAll(
+				'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])',
+			),
+		) as HTMLElement[]
+
+		const filteredElements: HTMLElement[] = []
+
+		elements.forEach((element) => {
+			if (
+				element.hasAttribute('disabled')
+				|| element.getAttribute('aria-hidden')
+			) {
+				return
+			}
+
+			filteredElements.push(element)
+		})
+
+		return filteredElements
+	}
```

Comment:
Est ce qu'on ne pourrai pas simplifier: async function getSelectableElements(): Promise<HTMLElement[]> {
	const parentNode = dialogContent?.value?.$el; // Is undefined when dialog is closed
	if (!parentNode) {
		return [];
	}
	return Array.from(
		parentNode.querySelectorAll(
			'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])',
		)
	).filter(
		(element) =>
			!element.hasAttribute('disabled') &&
			!element.getAttribute('aria-hidden')
	) as HTMLElement[];
} le filter permet d'eviter une boucle et la variable intermediaire